src/main/java/plugily/projects/villagedefense/kits/premium/NakedKit.java

```diff
@@ -104,17 +105,30 @@ public void onArmor(InventoryClickEvent event) {
     if(!(getPlugin().getUserManager().getUser(who).getKit() instanceof NakedKit)) {
       return;
     }
+    ClickType clickType = event.getClick();
+    if (clickType == ClickType.DROP || clickType == ClickType.CONTROL_DROP) {
+      return;
+    }
     Inventory inventory = event.getClickedInventory();
     if(inventory == null || inventory.getType() != InventoryType.PLAYER) {
       return;
     }
+
+    boolean hasArmor = false;
     ItemStack itemStack = event.getCurrentItem();
-    if(itemStack == null || !armorTypes.contains(itemStack.getType())) {
-      return;
+    if(itemStack != null && armorTypes.contains(itemStack.getType())) {
+      hasArmor = true;
+    }
+    if (clickType == ClickType.NUMBER_KEY) {
```

Comment:
I think the result will be the same if you put here an `else if` branch. So the `hasArmor` property will not change to `true` again if the first condition returns true.